cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowUtil.qll

```diff
@@ -680,12 +682,16 @@ VariableAccess getAnAccessToAssignedVariable(Expr assign) {
  *
  * It is important that all extending classes in scope are disjoint.
  */
-class BarrierGuard extends Expr {
-  /** NOT YET SUPPORTED. Holds if this guard validates `e` upon evaluating to `branch`. */
-  abstract deprecated predicate checks(Expr e, boolean branch);
+class BarrierGuard extends GuardCondition {
+  /** Override this predicate to hold if this guard validates `e` upon evaluating to `b`. */
```

Comment:
`b` -> `branch` (or vice versa)

Reply:
Good catch. I had trouble with the naming of this Boolean since there's a _field_ named `branch` on the `IRGuardCondition` class, and parameters are not allowed to shadow fields. I've named it `b` so it's consistent between IR and non-IR.